datasources/src/test/scala/com/normation/plugins/datasources/UpdateHttpDatasetTest.scala

```diff
@@ -1026,16 +986,16 @@ class UpdateHttpDatasetTest extends Specification with BoxSpecMatcher with Logga
   * - an array with one value (whatever the value) => remove array, get value as JSON
   *   - so an array of array will return an array
   * - an array with several values => keep array
-  */ 
+  */
   // testing arrays, array of arrays, etc
   lazy val testArray = List(
     // [] is a special case handled elsewhere
     // what API give                             -   expected with [*]                          - expected with [:1]
     ("""[ 1 ]"""                                 , """1"""                                      , """1"""                                 )
-  , ("""[ "a" ]"""                               , """"a""""                                    , """"a""""                               )
+  , ("""[ "a" ]"""                               , """a"""                                      , """a"""                                 )
   , ("""[ { "a" : 1 }]"""                        , """{"a":1}"""                                , """{"a":1}"""                           )
-  , ("""[ 1, 2 ]"""                              , """[1,2]"""                                  , """1"""                                 ) // array of size 1 are lifted
-  , ("""[ "a", "b" ]"""                          , """["a","b"]"""                              , """"a""""                               ) // array of size 1 are lifted
+  , ("""[ 1, 2 ]"""                              , """["1","2"]"""                              , """1"""                                 ) // array of size 1 are lifted
```

Comment:
this one... I'm not sure we want to keep it like that. Our top object are only either json object, json arrays, or json string. Here, we look at each int, transform it in a valid top object (so a string), and put it back in an array. I'm pretty sure it will breaks things, but I'm not sure how to deal with that.

Reply:
I think we should keep the type we have (int, boolean)

Maybe we should look how the agent treats int / boolean when read from a json property

Reply:
So we could have the more suitable format for this data

Reply:
I'm pretty sure we should not transform int to string. Why can't we have a top object json int ?

Reply:
anyway for now I'll just merge it and we need to think what we want to achieve here (and it will be on Rudder / properties side)

Reply:
@ncharles that would break compat. Historically, we only had string, but without quote: if an user put `1`, internally for us it's the string `1`, but it's outputed as `1` whitout quotes. It means that we don't put semantic besides string for that, and it's because we don't understand anything but string for directive parameters. 
We added JSON object and array at top level, and in the only case where we do extraction for the user (access of properties in `${node.properties[]}`), we take care to remove quotes on string. 

I have an idea to not break it.